src/main/java/training/endava/skeleton/Company.java

```diff
@@ -0,0 +1,53 @@
+package training.endava.skeleton;
+
+import java.time.Year;
+import java.util.List;
+
+public class Company {
+    private String name;
+    private List<Person> employees;
+    private int foundationYear;
+    private CompanyArea area;
+
+    public Company(String name, List<Person> employees, int foundationYear, CompanyArea area){
+        this.name = name;
+        this.employees = employees;
+        if(foundationYear > 1800 && foundationYear < 2018)
```

Comment:
As a best practice, you want to extract the validation values as constants of the class: MIN_YEAR_VALUE or something similar. And for this use case, you probably want to get the current year as the MAX_YEAR_VALUE so that there's no need to change it every year. Maybe use the Java8 `LocalDate` ?

Reply:
There was a task to work on primitive streams, but I had no primitive field, so I found appropiate to use the year as a primitive.